### DISCUSSION TITLE
Revert to the initial code after cancelling a completion

### DISCUSSION THREAD

**Discussion by bjorn-ove** [helix-editor/helix#7876] [General] [answered]
There are several reasons for why you would want to cancel a completion.

Sometimes it pops up where it is not wanted, such as when you are scrolling past something and go a little slow. In this case you notice that your down arrow move in the completion rather than the buffer and my natural instinct is to hit escape to abort the completion.

Sometimes you don't find what you are looking for and simply want to abort, thus hit escape.

In both these cases you end up with altered code, sometimes across multiple lines.

The expected behavior in my opinion is to revert back to how things where before the completion popup.

I might look into solving this myself, but wanted to get some input on whether there are good reasons for the existing behavior first.

**Comment by the-mikedavis** [comment]
If you're looking to cancel the completion and revert to the code before the completion started you should cancel with `C-c` rather than `esc`

**Comment by bjorn-ove** [comment]
I see, that does make some sense I guess, but what is the use-case for esc not being the same as C-c in this case?

**Comment by pascalkuthe** [comment, marked as answer]
When you select a completion (with `tab`) it gets immediately applied. `esc` will just close the popup `c-c` will undo the completion. So the usecase is closing the menu without undoing the completion

**Reply by bjorn-ove** [reply]
Oh, when I want that behavior I hit `return` instead. I notice a small difference. When I hit `return` it shows the documentation for parameters, while when I hit escape it doesn't.

So I guess this is intended behavior then. I tried to find documentation about it, but as far as I can tell the entire completion popup is not really documented when it comes to keyboard shortcuts.

Any way I can override `esc` to behave like `C-c` in this case, like it does for most others?

**Reply by pascalkuthe** [reply]
no see #5505, yes hitting ret does alomst the same thing but te reason its implemented this way is for consistency: esc always closes the popup but doesn't do anything beyond that while `c-c` cancels
